javatests/arcs/core/analysis/RecipeGraphFixpointIteratorTest.kt

```diff
@@ -0,0 +1,267 @@
+package arcs.core.analysis
+
+import arcs.core.data.HandleConnectionSpec
+import arcs.core.data.HandleMode
+import arcs.core.data.ParticleSpec
+import arcs.core.data.Recipe
+import arcs.core.data.TypeVariable
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import org.junit.Assert.assertTrue
+import org.junit.Assert.assertNotNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+/** An abstract domain of set, where the elements are ordered by inclusion. */
+class AbstractSet<S>(
```

Comment:
Might be worth putting this in its own file, I can see it being useful in other test files too. (And then it should live in a testing package under //java, not //javatests)

Reply:
Yes, this can be useful for more than tests. Will move this out in a separate PR.